src/pages/workspaces/workspace/JobHistory.js

```diff
@@ -128,183 +125,188 @@ const JobHistory = _.flow(
-          })
-        ]),
-        !loading && !hasJobs && noJobsMessage,
-        aborting && h(Modal, {
-          onDismiss: () => this.setState({ aborting: undefined }),
-          title: 'Abort All Workflows',
-          showX: true,
-          okButton: () => {
-            Ajax().Workspaces.workspace(namespace, name).submission(aborting).abort()
-              .then(() => this.refresh())
-              .catch(e => this.setState({ loading: false }, () => reportError('Error aborting submission', e)))
-            this.setState({ aborting: undefined, loading: true })
+            },
+            {
+              size: { basis: 250, grow: 0 },
+              headerRenderer: () => h(HeaderCell, ['Data entity']),
+              cellRenderer: ({ rowIndex }) => {
+                const { submissionEntity: { entityName, entityType } = {} } = filteredSubmissions[rowIndex]
+                return h(TooltipCell, [entityName && `${entityName} (${entityType})`])
+              }
+            },
+            {
+              size: { basis: 170, grow: 0 },
+              headerRenderer: () => h(HeaderCell, ['No. of Workflows']),
+              cellRenderer: ({ rowIndex }) => {
+                const { workflowStatuses } = filteredSubmissions[rowIndex]
+                return h(TextCell, Utils.formatNumber(_.sum(_.values(workflowStatuses))))
+              }
+            },
+            {
+              size: { basis: 150, grow: 0 },
+              headerRenderer: () => h(HeaderCell, ['Status']),
+              cellRenderer: ({ rowIndex }) => {
+                const { workflowStatuses, status } = filteredSubmissions[rowIndex]
+                return h(Fragment, [
+                  statusCell(workflowStatuses), _.keys(collapsedStatuses(workflowStatuses)).length === 1 && status
+                ])
+              }
+            },
+            {
+              size: { min: 220, max: 220 },
+              headerRenderer: () => h(HeaderCell, ['Actions']),
+              cellRenderer: ({ rowIndex }) => {
+                const {
+                  methodConfigurationNamespace, methodConfigurationName, methodConfigurationDeleted, submissionId, workflowStatuses,
+                  status, submissionEntity
+                } = filteredSubmissions[rowIndex]
+                return h(Fragment, [
+                  (!isTerminal(status) && status !== 'Aborting') && h(ButtonPrimary, {
+                    onClick: () => setAbortingId(submissionId)
+                  }, ['Abort workflows']),
+                  isTerminal(status) && (workflowStatuses['Failed'] || workflowStatuses['Aborted']) &&
+                  submissionEntity && !methodConfigurationDeleted && h(ButtonPrimary, {
+                    onClick: () => rerunFailures({
+                      workspace,
+                      submissionId,
+                      configNamespace: methodConfigurationNamespace,
+                      configName: methodConfigurationName,
+                      onDone: refresh
```

Comment:
TOL: It looks like there may be some pre-existing issues here, since this will kick off a separate refresh loop without stopping the first one. Not sure if it's worth fixing now, but we should probably do something like the `reschedule` pattern in `useCloudEnvironmentPolling`.

Reply:
What if I just add this to the top of the refresh function itself?
```js
    if (scheduledRefresh.current) {
      clearTimeout(scheduledRefresh.current)
    }
```